cmd/base.go

```diff
@@ -29,6 +30,9 @@ var (
 	// Commit is the git commit used to build the app. Designed to be overwritten by make.
 	Commit string
 
+	// Mu is the mutex for accessing global variables in this package.
+	Mu sync.RWMutex
```

Comment:
i don't like introducing global mutex. it was a mistake to introduce global context and cancel function in the first place, maybe it is possible to refactor code to avoid global context? or it is too many changes?

Reply:
not too many changes, fixed